### PR TITLE
[PLATFORM-3258] Update CLI to algo rust version 3 crate. Switch to RustTLS

### PR DESCRIPTION
**Pull request by bytekast**

### CODE REVIEW

**Review by anowell — APPROVED** [review]
I'll recommend a PR that removes `runlocal` from https://github.com/algorithmiaio/dev-center/blob/8bcfa8d609a985ec5687e050b8b38675d0f85337/_pages/algorithm-development/your-first-algo.md#L90

**Comment by bytekast** [comment]
@anowell removed the custom OpenSSL build script

**Comment by anowell** [comment]
I believe the docker build approach was also being used to build against an old libc (for maximum compatibility with older distros). I think that buildslave image was a centos5 docker image where now it will build on Travis's Ubuntu Trusty. Centos 5 is no longer supported, but I think it might be worth doing the build still on a Centos6 host. See: https://github.com/algorithmiaio/algorithmia-cli/issues/6

Everything else here I agree with.

File: .travis.yml [hunk 1]
```diff
@@ -12,32 +12,29 @@ matrix:
...
            TARGET=x86_64-apple-darwin
            MACOSX_DEPLOYMENT_TARGET=10.7
+           FEATURES=rust-tls
     - os: linux
       env: RUST_VERSION=stable
            TARGET=x86_64-unknown-linux-gnu
-           DOCKER=alexcrichton/rust-slave-dist:2016-11-15
+           DOCKER_BUILD=yes
            MAKE_DEB=yes
+           FEATURES=default
```
Comment (anowell) [inline .travis.yml]: Isn't this backward? This is doing rust-tls for OSX and native-tls for Linux (which means openssl)?

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,31 @@
+FROM centos:centos6
+MAINTAINER Rowell Belen "rowell.belen@bytekast.com"
+
+RUN yum -y update && yum clean all
+RUN yum -y install git gcc gcc-c++ make curl openssl openssl-devel ca-certificates tar && yum clean all
```
Comment (anowell) [inline Dockerfile:5]: Thought this was moving away from openssl on Linux. I'm guessing openssl is needed for curl and cargo to download stuff, but openssl-devel?

**Review by anowell — COMMENTED** [review]
Overall, I'm good with this PR, but just those 2 minor comments for things tripping me up. Also, I think the windows build is produced by AppVeyor, and that build is failing (looks like the appveyor config is just out of date - looking at it now)

**Pull request merged**